Kudu.Core/Deployment/DeploymentManager.cs

```diff
@@ -617,7 +617,15 @@ private DeployResult GetResult(string id, string activeDeploymentId, bool isDepl
                     {
                         await builder.Build(context);
                         builder.PostBuild(context);
-                        await _autoSwapHandler.HandleAutoSwap(id, context);
+
+                        // There are two code path will trigger autoswap (if enabled), Local git and Webhook
+                        // For Local git we would like to trigger autoswap once deployment is done
+                        // For Webhook, there might be a sequence of trigger, we only want to perform autoswap when last deployment is done, and this can only handle outside of the deployment
+                        // Hence "swapAfterDeploy" flag to control whether to perform swap within a deployment
+                        if (swapAfterDeploy)
```

Comment:
This "if" check is for local git, where deployment trigger by post-receive hook.  

I am thinking if we should move "_autoSwapHandler.HandleAutoSwap" outside of DeploymentManger.  Just as what I did in this PR, for the case of FetchHandler, perform autoswap after deployment process. Then we will not need this if statement and "swapAfterDeploy" property